api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java

```diff
@@ -53,18 +62,57 @@
         return ResponseEntity.ok(criteriaResponse);
     }
 
-    protected QueryResult getQueryResult(String type, Long parentId) {
-        BigQuery bigquery =
-                new BigQueryOptions.DefaultBigqueryFactory().create(BigQueryOptions.getDefaultInstance());
+    @Override
+    public ResponseEntity<SubjectListResponse> searchSubjects(SearchRequest request) {
+        QueryRequest query;
+        QueryResult result;
+        Map<String, Integer> resultMapper;
+
+        SubjectListResponse subjectSet = new SubjectListResponse();
+
+        if (request.getType().equals("ICD9")) {
+            List<SearchParameter> params = request.getSearchParameters();
+
+            /*  Check for SearchParameters without Domain IDs.
```

Comment:
If a user selects a parent node, the domain id is undefined - domain ids are only defined for leaf nodes.  I.e. the user can select an intermediate category, each entry of which may have a different domain id.  In that case, we go out and grab all the children of that parent node and resolve them to search parameters with domain ids.

Reply:
Can you give an example query for when that would happen? (I'm trying to understand what that means...)

Reply:
<img width="1555" alt="screen shot 2017-09-21 at 9 35 24 am" src="https://user-images.githubusercontent.com/31938607/30701611-52398ef8-9eb0-11e7-855c-16c532cadd3c.png">

If you look at the highlighted parent, you'll see that it is selectable - shorthand for selecting all the children.  Each child could have a different domain ID.  So this would generate a SearchParameter for the parent that needs to be resolved to a set of SearchParameters for the children and send _that_ back to the server.  The set of children, btw, is only loaded into the browser when the user actually opens that branch of the tree, so the information necessary to resolve the parent in the browser isn't present.

Reply:
Does this look only for direct children, or all descendants? How many children might be selected by this?

Reply:
Discussed offline. Should only have one level of hierarchy (parents of leaves.) We should do some testing to make sure this works OK for big parents.